thunderbird-unwrapped-91.5.1: undefined symbol wl_proxy_marshal_flags

### Describe the bug
Thunderbird fails to compile with "ld.lld: error: undefined symbol: wl_proxy_marshal_flags" when rebuilding system configuration.

### Steps To Reproduce
Steps to reproduce the behavior:
1. I updated my system configuration flake inputs (`nix flake update ~/src/nixos-configuration`)
2. ...and run system rebuild (`sudo nixos-rebuild switch --flake ~src/nixos-configuration`)

### Expected behavior
Thunderbird compiles successfully.

### Additional context

Relevant lines from the build log:

```text
/nix/store/ljiziif2pj26xpprqaf4c3x7qk88dlfz-clang-wrapper-13.0.0/bin/clang++ -std=gnu++17 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -fstack-protector-strong -fstack-clash-protection -Qunused-arguments -Qunused-arguments -Wall -Wbitfield-enum-conversion -Wdeprecated-this-capture -Wempty-body -Wformat-type-confusion -Wignored-qualifiers -Wpointer-arith -Wshadow-field-in-constructor-modified -Wshadow-uncaptured-local -Wsign-compare -Wtype-limits -Wno-error=tautological-type-limit-compare -Wunreachable-code -Wunreachable-code-return -Wno-invalid-offsetof -Wclass-varargs -Wempty-init-stmt -Wfloat-overflow-conversion -Wfloat-zero-conversion -Wloop-analysis -Wno-range-loop-analysis -Wc++2a-compat -Wcomma -Wenum-compare-conditional -Wimplicit-fallthrough -Wstring-conversion -Wno-inline-new-delete -Wno-error=deprecated-declarations -Wno-error=array-bounds -Wno-error=backend-plugin -Wno-error=free-nonheap-object -Wno-error=return-std-move -Wno-error=atomic-alignment -Wno-error=deprecated-copy -Wno-error=unused-but-set-variable -Wformat -Wformat-security -Wno-gnu-zero-variadic-macro-arguments -Wno-psabi -Wno-unknown-warning-option -fno-sized-deallocation -fno-aligned-new -fno-exceptions -fno-strict-aliasing -fPIC -fno-rtti -ffunction-sections -fdata-sections -fno-exceptions -fno-math-errno -pthread -pipe -g -O2 -fomit-frame-pointer -funwind-tables  -shared -Wl,-z,defs -Wl,--gc-sections -Wl,-h,libxul.so -o libxul.so /build/thunderbird-91.5.1/obj-x86_64-pc-linux-gnu/toolkit/library/build/libxul_so.list  -flto=thin -Wl,-plugin-opt=-import-instr-limit=10 -Wl,-plugin-opt=new-pass-manager -Wl,-plugin-opt=-import-hot-multiplier=30 -lpthread -fuse-ld=lld -Wl,-z,noexecstack -Wl,-z,text -Wl,-z,relro -Wl,-z,nocopyreloc -Wl,-Bsymbolic-functions -Wl,--build-id=sha1 -Wl,--icf=safe -fstack-protector-strong -fstack-clash-protection -Wl,-rpath-link,/build/thunderbird-91.5.1/obj-x86_64-pc-linux-gnu/dist/bin -Wl,-rpath-link,/nix/store/xsb6042lpvs6v1n97g5ci1x3bwy6cids-thunderbird-unwrapped-91.5.1/lib  ../../../js/src/build/libjs_static.a /build/thunderbird-91.5.1/obj-x86_64-pc-linux-gnu/x86_64-unknown-linux-gnu/release/libgkrust.a ../../../security/sandbox/linux/libmozsandbox.so ../../../comm/ldap/c-sdk/libraries/libldap/libldap60.so ../../../comm/ldap/c-sdk/libraries/libldif/libldif60.so ../../../comm/ldap/c-sdk/libraries/libprldap/libprldap60.so ../../../config/external/lgpllibs/liblgpllibs.so ../../../config/external/sqlite/libmozsqlite3.so ../../../widget/gtk/mozgtk/libmozgtk.so ../../../widget/gtk/mozwayland/libmozwayland.so -Wl,--version-script,symverscript  -ldl  -L/nix/store/w5z6skcz71hqq2kffflklq6shlwqrk2r-icu4c-70.1/lib -licui18n -licuuc -licudata -lpthread -ldl -lc -L/nix/store/8jxghs14jgj3ncrf53dij0wjdrqmc0wl-nspr-4.33/lib -lplds4 -lplc4 -lnspr4 -L/nix/store/ljr3c02zwg5sy1jcabjymjdyw7w3jrwa-libffi-3.4.2/lib/../lib64 -lffi -L/nix/store/92wmg8niqrvwxsfigh5hhjpb0y9ska1i-zlib-1.2.11/lib -lz -lm -L/nix/store/gsrjqz7wx7p5718yr95i6n9z35z3lsli-nss-3.74/lib64 -lssl3 -lsmime3 -lnss3 -lnssutil3 -L/nix/store/06yix2p8krcma80bx3c86408ns0lkahh-freetype-2.11.1/lib -lfreetype -L/nix/store/7vz5kf7g2appfgfwian8r59v1mfhi6sk-fontconfig-2.13.94-lib/lib -lfontconfig -L/nix/store/gm9zbw0aanbx1jq6rwjmzd9d5h62kkrc-gtk+3-3.24.31/lib -L/nix/store/jc0xdrv2zwhkj4mdkj3jj85ljiwm9dsn-atk-2.36.0/lib -L/nix/store/dz9ai5i2xs58kjp093bw2jq60d2rl63r-glib-2.70.2/lib -L/nix/store/4w78baviivpyl61ajzgljjm4dvl1pwm1-cairo-1.16.0/lib -L/nix/store/43m1nr6hbrz09ksgx6f2ncwa6n2kf8b0-gdk-pixbuf-2.42.6/lib -L/nix/store/ihqs85a1559xm6mv4lx8sin5pvzpjsv5-pango-1.50.3/lib -L/nix/store/vnycg9yjjmlfm14dnylzk4jq233snz73-harfbuzz-3.1.2/lib -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lrt -lXrender -lpng -L/nix/store/zs4vqwcahq0jmd35kamrk47srwaizzng-libevent-2.1.12/lib -levent -L/nix/store/j5jbhyd1dkmfqw8ld534pjn69b6al33x-libvpx-1.11.0/lib -lvpx -L/nix/store/abm1qqpkpi9jj5nyzz0d2jccmw1jv9yd-pixman-0.38.4/lib -lpixman-1 -L/nix/store/jc3siq1jzlbs5r588bcvm2q0x8p6ak4l-dbus-1.12.20-lib/lib -L/nix/store/ry2k6cfr2njyx2h9k031a0b7hmcn1mp0-dbus-glib-0.112/lib -ldbus-glib-1 -ldbus-1 -lX11-xcb -lxcb-shm -lxcb -lX11 -lXext -lpangoft2-1.0 -lXt -lgthread-2.0
ld.lld: error: undefined symbol: wl_proxy_marshal_flags
>>> referenced by wayland-client-protocol.h:3672 (/nix/store/01rcndpymayvrpswqx34znwfxx82bwvd-wayland-1.20.0-dev/include/wayland-client-protocol.h:3672)
>>>               lto.tmp:(mozilla::gl::DeleteWaylandGLSurface(void*))
>>> referenced by wayland-client-protocol.h:1291 (/nix/store/01rcndpymayvrpswqx34znwfxx82bwvd-wayland-1.20.0-dev/include/wayland-client-protocol.h:1291)
>>>               lto.tmp:(mozilla::gl::GLContextEGL::CreateEGLPBufferOffscreenContextImpl(std::shared_ptr<mozilla::gl::EglDisplay>, mozilla::gl::GLContextCreateDesc const&, mozilla::gfx::IntSizeTyped<mozilla::gfx::UnknownUnits> const&, bool, nsTSubstring<char>*))
>>> referenced by wayland-client-protocol.h:3914 (/nix/store/01rcndpymayvrpswqx34znwfxx82bwvd-wayland-1.20.0-dev/include/wayland-client-protocol.h:3914)
>>>               lto.tmp:(mozilla::layers::NativeLayerRootWayland::EnsureHideLayer(RefPtr<mozilla::layers::NativeLayerWayland> const&))
>>> referenced 85 more times
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [/build/thunderbird-91.5.1/config/rules.mk:545: libxul.so] Error 1
make[3]: Leaving directory '/build/thunderbird-91.5.1/obj-x86_64-pc-linux-gnu/toolkit/library/build'
make[2]: *** [/build/thunderbird-91.5.1/config/recurse.mk:72: toolkit/library/build/target] Error 2
make[2]: Leaving directory '/build/thunderbird-91.5.1/obj-x86_64-pc-linux-gnu'
make[1]: *** [/build/thunderbird-91.5.1/config/recurse.mk:34: compile] Error 2
make[1]: Leaving directory '/build/thunderbird-91.5.1/obj-x86_64-pc-linux-gnu'
make: *** [/build/thunderbird-91.5.1/config/rules.mk:355: default] Error 2
```

So it looks like the error has something to do with Wayland support.

### Notify maintainers
@eelco @lovesegfault @pierron @vcunat

### Metadata

nixpkgs lock:

```json
{
  "lastModified": 1644398404,
  "narHash": "sha256-E/NXWUjHUyAS64JvQZJISfnC1jti8IQ9Ks5qynToL1s=",
  "owner": "NixOS",
  "repo": "nixpkgs",
  "rev": "c9fd5b70d716fcbbb2964fc15239b34d816ecca1",
  "type": "github"
}
```

nix-info:

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.93, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.7.0pre20220124_0a70b37`
 - channels(root): `"nixos-21.05.1076.bad3ccd099e"`
 - channels(nameless): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/applications/networking/mailreaders/thunderbird/packages.nix
+++ b/pkgs/applications/networking/mailreaders/thunderbird/packages.nix
@@ -10,12 +10,12 @@ in
 rec {
   thunderbird = common rec {
     pname = "thunderbird";
-    version = "91.5.1";
+    version = "91.6.0";
     application = "comm/mail";
     binaryName = pname;
     src = fetchurl {
       url = "mirror://mozilla/thunderbird/releases/${version}/source/thunderbird-${version}.source.tar.xz";
-      sha512 = "5939e09b143e440efa906d95cda06826bd3a73b2edde5eed86229b8a0e4d1434519059f37d319d26978d7eea9b3906c5e1c1543a2bc2465625d5ab5438855717";
+      sha512 = "a11eafe1390141ee3508eea06ba8ab135d0725513977a3b37b3b35f413a1f825dc14fef530b9ac961840804be59291c7f5cba3c93b12726605d4a7255660f749";
     };
     patches = [
       # The file to be patched is different from firefox's `no-buildconfig-ffx90.patch`.

